compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion/Primitives.hs

```diff
@@ -44,15 +44,15 @@ convertPrim _ "SGetParty" (t1@TText :-> TScenario TParty) =
     ETmLam (varV1, t1) $ EScenario $ SGetParty $ EVar varV1
 
 -- Comparison
-convertPrim v "BEEqual" (a1 :-> a2 :-> TBool) | a1 == a2, v `supports` featureGenericComparison =
+convertPrim _ "BEEqual" (a1 :-> a2 :-> TBool) | a1 == a2 =
     EBuiltin BEEqualGeneric `ETyApp` a1
-convertPrim v "BELess" (a1 :-> a2 :-> TBool) | a1 == a2, v `supports` featureGenericComparison =
+convertPrim _ "BELess" (a1 :-> a2 :-> TBool) | a1 == a2 =
     EBuiltin BELessGeneric `ETyApp` a1
-convertPrim v "BELessEq" (a1 :-> a2 :-> TBool) | a1 == a2, v `supports` featureGenericComparison =
+convertPrim _ "BELessEq" (a1 :-> a2 :-> TBool) | a1 == a2 =
     EBuiltin BELessEqGeneric `ETyApp` a1
-convertPrim v "BEGreater" (a1 :-> a2 :-> TBool) | a1 == a2, v `supports` featureGenericComparison =
+convertPrim _ "BEGreater" (a1 :-> a2 :-> TBool) | a1 == a2 =
     EBuiltin BEGreaterGeneric `ETyApp` a1
-convertPrim v "BEGreaterEq" (a1 :-> a2 :-> TBool) | a1 == a2, v `supports` featureGenericComparison =
+convertPrim _ "BEGreaterEq" (a1 :-> a2 :-> TBool) | a1 == a2 =
     EBuiltin BEGreaterEqGeneric `ETyApp` a1
 convertPrim _ "BEEqual" (TBuiltin a1 :-> TBuiltin a2 :-> TBool) | a1 == a2 =
```

Comment:
I think we can remove this case and up to `convertPrim _ "BEGreater" (TBuiltin a1 :-> TBuiltin a2 :-> TBool)` below since they're already covered by the cases above

Reply:
good catch d43406131fe916d3a102a8ebeb3746070164a1a7

Reply:
Ah that's a different set of cases, I meant the set with `TBuiltin a1 :-> TBuiltin a2 :-> TBool` as the type. d43406131fe916d3a102a8ebeb3746070164a1a7 drops the cases with type `TNumeric10 :-> TNumeric10 :-> TBool`

Reply:
so much stuff to delete 2e757a78d6ff20d7fda45fc7c59a886d8db4d027 I think we can also kill the `BEEqual` and friends constructors in favor of the generic versions.

Reply:
will leave that for a separate PR.

Reply:
that sounds good! thanks :)